src/snowflake/connector/pandas_tools.py

```diff
@@ -35,6 +35,9 @@
 
 def chunk_helper(lst: T, n: int) -> Iterator[tuple[int, T]]:
     """Helper generator to chunk a sequence efficiently with current index like if enumerate was called on sequence."""
+    if n >= len(lst):
+        yield 0, lst
+        return
```

Comment:
So it worked for shorter lists already:
```
>>> list(chunk_helper([1, 2], 3))
[(0, [1, 2])]
```
There's only one case the original function didn't work for and so this is more explicit:
```suggestion
    if n == len(lst):
        yield 0, lst
        return
```